### PR TITLE
refactor: Pull down to refresh for the MyApps page and the Home page

### PR DESCRIPTION
**Pull request by liiil825**

### CODE REVIEW

**Comment by liiil825** [comment]
#587

File: src/components/InfiniteScroll/index.jsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    useWindow: PropTypes.bool
+  };
+
+  static defaultProps = {
+    className: '',
+    hasMore: false,
+    loadMore: _.noop,
+    initialLoad: false,
+    isLoading: true,
+    loader: <Loading isLoading={true} />,
```
[the remark this answers is not in the document]
Comment (liiil825) [reply src/components/InfiniteScroll/index.jsx]: ok

File: src/pages/Dashboard/Apps/MyApps/index.jsx [hunk 2]
```diff
@@ -32,13 +33,15 @@ export default class Apps extends Component {
   async componentDidMount() {
     const { appStore, appVersionStore, user } = this.props;
     const { user_id } = user;
-    appStore.pageSize = 200;
+
+    appStore.pageSize = 48;
```
Comment (iwisunny) [inline src/pages/Dashboard/Apps/MyApps/index.jsx]: You can set `pageSize` in store file, not in component

File: src/pages/Dashboard/Apps/MyApps/index.jsx [hunk 3]
```diff
@@ -101,9 +104,16 @@ export default class Apps extends Component {
       <Layout className={styles.layout} noSubMenu={true}>
         <Loading className={styles.page} isLoading={pageLoading}>
           {this.renderHeader()}
-          <div className={styles.cards}>
-            {apps.map(item => <Card key={item.app_id} t={t} data={item} />)}
-          </div>
+          <InfiniteScroll
+            pageStart={appStore.currentPage}
```
Comment (iwisunny) [inline src/pages/Dashboard/Apps/MyApps/index.jsx]: Why not extract variables from appStore?

File: src/pages/Dashboard/Apps/MyApps/index.jsx [hunk 4]
```diff
@@ -101,9 +104,16 @@ export default class Apps extends Component {
...
+            loadMore={appStore.loadMore}
+            isLoading={appStore.isLoading}
+            hasMore={appStore.hasMore}
+          >
+            <div className={styles.cards}>
+              {apps.map(item => <Card key={item.app_id} t={t} data={item} />)}
```
Comment (iwisunny) [inline src/pages/Dashboard/Apps/MyApps/index.jsx]: What does `t` mean ?

File: src/pages/Home/index.jsx [hunk 5]
```diff
@@ -19,6 +20,13 @@ import styles from './index.scss';
 }))
 @observer
 export default class Home extends Component {
+  constructor(props) {
```
Comment (iwisunny) [inline src/pages/Home/index.jsx]: No need costructor, just 
```
state ={pageLoading: true}
```

File: src/pages/Home/index.jsx [hunk 6]
```diff
@@ -43,15 +50,16 @@ export default class Home extends Component {
...
     }
 
-    if (category) {
-      filterParams.category_id = category;
-    }
-    if (search) {
-      filterParams.search_word = search;
-    }
-    await appStore.fetchAll(filterParams);
+    appStore.selectStatus = 'active';
```
Comment (iwisunny) [inline src/pages/Home/index.jsx]: refactor with Object.assign

Comment (liiil825) [reply src/pages/Home/index.jsx]: ok

File: src/pages/Home/index.jsx (hunk 5 above)
Comment (liiil825) [reply src/pages/Home/index.jsx]: ok

File: src/pages/Dashboard/Apps/MyApps/index.jsx (hunk 4 above)
Comment (liiil825) [reply src/pages/Dashboard/Apps/MyApps/index.jsx]: the function from i18n

File: src/pages/Dashboard/Apps/MyApps/index.jsx (hunk 3 above)
Comment (liiil825) [reply src/pages/Dashboard/Apps/MyApps/index.jsx]: ok

**Pull request merged**